optimum/intel/openvino/trainer.py

```diff
@@ -719,8 +719,10 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
                     if not save_as_external_data:
                         onnx_path.write_bytes(f.getvalue())
                     logger.warning(
-                        f"Error encountered during IR pruning: {err}. {onnx_path} is dumped for debug. Run continues."
+                        f"Error encountered during OpenVINO IR pruning: {err}. {onnx_path} is dumped for debugging. "
+                        "Run continues without saving OpenVINO IR files."
                     )
+                    return
             else:
                 if self._get_compression_controller_by_cls(QuantizationController) is not None:
```

Comment:
It's a blanket exception, it's a bit risky to only show a warning. Shouldn't we reraise the exception and then still dump the ONNX model? Then it's clear that something is broken, and you get the exact error

Reply:
> It's a blanket exception, it's a bit risky to only show a warning. Shouldn't we reraise the exception and then still dump the ONNX model? Then it's clear that something is broken, and you get the exact error

I agree with your concern. See updates: https://github.com/huggingface/optimum-intel/blob/5618c3cf45d61bab48d702a0254001395e66f182/optimum/intel/openvino/trainer.py#L717-L724